File: codeforces/677B/cpp/677b.cpp
#include <cstdio>

long long an[100001];
int main()
{
    long long n, h, k;
    scanf("%lld %lld %lld", &n, &h, &k);
    for (long long i = 0; i < n; i++) {
        scanf("%lld", &an[i]);
    }

    long long ans = 0, i = 0, height = 0;
    for (; i < n; i++) {
        if (height + an[i] <= h) {
            height += an[i];
        }
        else {
            ans++;
            height = an[i];
        }

        ans += height / k;
        height %= k;
    }

    if (height != 0) {
        ans += height / k;
        height %= k;
        if (height != 0) {
            ans++;
        }
    }

    printf("%lld\n", ans);

    return 0;
}

File: codeforces/677B/py/677b.py
import sys

n, h, k = map(int, sys.stdin.readline().split())
an = list(map(int, sys.stdin.readline().split()))

ans = 0
cur = 0
height = 0
while cur < n:
    if height + an[cur] <= h:
        height += an[cur]
    else:
        ans += 1
        height = an[cur]

    ans += height // k
    height %= k

    cur += 1
    if cur == n and height != 0:
        ans += 1

print(ans)
